x-pack/platform/plugins/shared/fleet/server/services/package_policy.ts

```diff
@@ -1117,24 +1118,25 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
     // Bump revision of all associated agent policies (old and new)
     const associatedPolicyIds = new Set([...oldPackagePolicy.policy_ids, ...newPolicy.policy_ids]);
     logger.debug(`Bumping revision of associated agent policies ${associatedPolicyIds}`);
-    const bumpPromises = [];
-    for (const policyId of associatedPolicyIds) {
-      // Check if the agent policy is in both old and updated package policies
-      const assignedInOldPolicy = oldPackagePolicy.policy_ids.includes(policyId);
-      const assignedInNewPolicy = newPolicy.policy_ids.includes(policyId);
-
-      // Remove protection if policy is unassigned (in old but not in updated) or policy is assigned (in updated but not in old)
-      const removeProtection =
-        (assignedInOldPolicy && !assignedInNewPolicy) ||
-        (!assignedInOldPolicy && assignedInNewPolicy);
-
-      bumpPromises.push(
-        agentPolicyService.bumpRevision(soClient, esClient, policyId, {
+    const bumpPromise = pMap(
+      associatedPolicyIds,
+      (policyId) => {
+        // Check if the agent policy is in both old and updated package policies
+        const assignedInOldPolicy = oldPackagePolicy.policy_ids.includes(policyId);
+        const assignedInNewPolicy = newPolicy.policy_ids.includes(policyId);
+
+        // Remove protection if policy is unassigned (in old but not in updated) or policy is assigned (in updated but not in old)
+        const removeProtection =
+          (assignedInOldPolicy && !assignedInNewPolicy) ||
+          (!assignedInOldPolicy && assignedInNewPolicy);
+
+        return agentPolicyService.bumpRevision(soClient, esClient, policyId, {
           user: options?.user,
           removeProtection,
-        })
-      );
-    }
+        });
+      },
+      { concurrency: MAX_CONCURRENT_EPM_PACKAGES_INSTALLATIONS }
```

Comment:
Probably could use the `MAX_CONCURRENT_AGENT_POLICIES_OPERATIONS` here as it's an agent policy operation not a package install